BACnetServerExample.py

```diff
@@ -84,386 +85,738 @@
+          propertyArrayIndex)
     return False
-def CallbackGetPropertyDouble(deviceInstance, objectType, objectInstance, propertyIdentifier, value, useArrayIndex, propertyArrayIndex):
-    print("CallbackGetPropertyDouble", deviceInstance, objectType, objectInstance, propertyIdentifier, useArrayIndex, propertyArrayIndex )
+
+
+def CallbackGetPropertyDouble(deviceInstance, objectType, objectInstance, propertyIdentifier, value, useArrayIndex,
+                              propertyArrayIndex):
+    print("CallbackGetPropertyDouble", deviceInstance, objectType, objectInstance, propertyIdentifier, useArrayIndex,
+          propertyArrayIndex)
     return False
-def CallbackGetPropertyOctetString(deviceInstance, objectType, objectInstance, propertyIdentifier, value, valueElementCount, maxElementCount, useArrayIndex, propertyArrayIndex):
-    print("CallbackGetPropertyOctetString", deviceInstance, objectType, objectInstance, propertyIdentifier, maxElementCount, useArrayIndex, propertyArrayIndex )
+
+
+def CallbackGetPropertyOctetString(deviceInstance, objectType, objectInstance, propertyIdentifier, value,
+                                   valueElementCount, maxElementCount, useArrayIndex, propertyArrayIndex):
+    print("CallbackGetPropertyOctetString", deviceInstance, objectInstance, propertyIdentifier, maxElementCount,
+          useArrayIndex, propertyArrayIndex)
+    if propertyIdentifier == bacnet_propertyIdentifier["ipaddress"]:
+        if objectType == bacnet_objectType["networkPort"] and objectInstance == db["networkPort"]["instance"]:
+            valueElementCount[0] = db["networkPort"]["ipLength"]
+            octetStringCopy(db["networkPort"]["ipAddress"], value, valueElementCount[0])
+            print(f"DEBUG: IN GET IP: output = {value[0]}.{value[1]}.{value[2]}.{value[3]}")
+            return True
+    elif propertyIdentifier == bacnet_propertyIdentifier["ipdefaultgateway"]:
+        if objectType == bacnet_objectType["networkPort"] and objectInstance == db["networkPort"]["instance"]:
+            valueElementCount[0] = db["networkPort"]["ipLength"]
+            octetStringCopy(db["networkPort"]["ipDefaultGateway"], value, valueElementCount[0])
+            return True
+    elif propertyIdentifier == bacnet_propertyIdentifier["ipsubnetmask"]:
+        if objectType == bacnet_objectType["networkPort"] and objectInstance == db["networkPort"]["instance"]:
+            valueElementCount[0] = db["networkPort"]["ipLength"]
+            octetStringCopy(db["networkPort"]["ipSubnetMask"], value, valueElementCount[0])
+            return True
+    elif propertyIdentifier == bacnet_propertyIdentifier["ipdnsserver"]:
+        if objectType == bacnet_objectType["networkPort"] and objectInstance == db["networkPort"]["instance"]:
+            valueElementCount[0] = db["networkPort"]["ipNumOfDns"] * 4
+            for i in range(db["networkPort"]["ipNumOfDns"]):
```

Comment:
Take the size of db["networkPort"]["ipDnsServer"] and use that to compute the ipNumOfDns